### DISCUSSION TITLE
Datable sorting + AJAX call

### DISCUSSION THREAD

**Discussion by timothydilbert** [carbon-design-system/carbon#15162] [Help]
I have a use case involving the DataTable component with sorting enabled. When the user attempts to sort the table via the table header, I want to trigger an AJAX call.

I'm able to capture on the `onClick` event, but figuring out whether I should be sorting ASC or DESC is difficult. And, the `sortRow` does not trigger for me. And, even if it did, it appears to be client-side ordering, which doesn't work for my use case.

I'm really just looking for an `onClick` event that includes the sort status (e.g. `arrow--up` or `arrow--down`). Is this possible, and how do I do that?

**Comment by John30013** [comment]
I've wanted to be able to do that too, but I've concluded that the DataTable component simply doesn't support it. I suppose you could attach your own click event listener on the table headings and probably examine a class name on the element when it's clicked to determine the sort direction. But you might be better off just creating your own DataTable component that works how you need it to.
